### ISSUE TITLE
Error updating sketch by FTDI (USB to serial adapter) - ATmega2561

### ISSUE THREAD

**Issue by janioanselmo** [MCUdude/MegaCore#88]
## OF THE PROBLEM:
I have not been able to update the microcontroller over the USB (serial to USB adapter) for a long time. I currently use the USBAsp recorder, and it is working perfectly. The problem is as a developer, no problems, but... this board will be in the hands of beginning users and sending the file through the Arduino IDE via USB is very practical. However, I've done everything I could check, and to this day I'm missing something I can not locate. Let's go to the steps already done, according to @MCUdude:

1. Hit Burn Bootloader. If an LED is connected to PB5/PB7 pin, it should flash twice every second;
    - **Succssesfully done!**
2. Then select the correct serial port under the Tools menu, and click the Upload button. If you are getting some kind of timeout error, it means your RX and TX pins are swapped, or your auto reset circuit is working properly (the 100 nF capacitor on the reset line)
    - **We perform the RX/TX inversion (to ensure they were not reversed).**

But when I click on: "**Upload butto**n" recording does not occur! I will pass the "**schematic.pdf**" and the used converters (try with two different devices)

## OF OBSERVATIONS:
- IDE version Arduino 1.8.5;
- Drivers installed correctly, in the Arduino IDE interface correctly recognizes the serial port (COM5). Echo test performed successfully;
- Selecting the recorder: "AVRISP mkII";
- I researched and did not find people with the same problem, before posting that question.

## OF ANNEXES:
![ftdi adapter ii](https://user-images.githubusercontent.com/32854172/39136197-0906d1ac-46f1-11e8-8627-99776c45fee2.jpg)
![ftdi adapter](https://user-images.githubusercontent.com/32854172/39136198-092aa05a-46f1-11e8-811a-9195f045daf2.jpg)
[Schematic.pdf](https://github.com/MCUdude/MegaCore/files/1939915/Schematic.pdf)

**Comment by MCUdude** [comment]
According to your schematic it seems like RX and TX is swapped. What's more strange is how you were able to program the ATmega2561 using the ISP header! If you look at the minimal setup schematic in the README, you'll see that MOSI and MISO on the ISP header is actually connected to PE0 and PE1

**Comment by janioanselmo** [comment]
First, thanks for the quick reply! Looking right, you're right. I ended up uploading the old/wrong schematic, I have already updated the file with the correct version (sorry). Just to remember: I successfully do USBAsp recording, however via UART/Serial no! And I already did the inversion of TX/R and RX/TX and I do not succeed! However, I do not know what else may have gone wrong.
Thank you, @MCUdude

**Comment by MCUdude** [comment]
Try placing two 1k resistors in series with the RX and TX line. Also, can you please post a photo of the actual board? It might be easier to spot errors

**Comment by janioanselmo** [comment]
Excellent, I will do this, and yes, I will post a photo readl of the board!

**Comment by MCUdude** [comment]
I designed this board a while back, and I've used it with ATmega2561. Have not experienced any issues when uploading using USB to serial. 

If you connect and LED to pin 13, select the proper settings in the boards meny and hit "Burn bootloader", the LED should now flash twice every second. This indicates that the bootloader is installed and running.

[MegaCore_Shematic.pdf](https://github.com/MCUdude/MegaCore/files/1940004/MegaCore_Shematic.pdf)
![2018-04-23 23 05 02](https://user-images.githubusercontent.com/3238730/39153295-fac06aae-474a-11e8-9c6d-a0f91150fde9.jpg)

**Comment by janioanselmo** [comment]
Hi, @MCUdude! Great project! I am grateful to share the schematic. I put a 1k resistor on the TX line and another 1k resistor on the RX line and follow the same problem! I am committing a very serious fault and I do not see!

**avrdude: stk500_recv (): programmer is not responding
avrdude: stk500_getsync () attempt 1 of 10: not in sync: resp = 0x2d
avrdude: stk500_recv (): programmer is not responding
avrdude: stk500_getsync () attempt 2 of 10: not in sync: resp = 0x2d
avrdude: stk500_recv (): programmer is not responding
avrdude: stk500_getsync () attempt 3 of 10: not in sync: resp = 0x2d**

**Comment by MCUdude** [comment]
Thanks!
Does the LED flash/D13 toggle after you've clicked **Burn Bootloader**. We need to find out if the bootloader is present or not..

**Comment by janioanselmo** [comment]
@MCUdude, Oh, some information that may be important: Whenever I press the "upload" it gives a "reset" to the card ... it makes the attempts to write (without success) and it ends with a "new" reset.

**Comment by janioanselmo** [comment]
Yes the bootloader is correct! So much, I can record with the USBasp recorder without problems! Or does not this guarantee us the bootloader installation?

**Comment by MCUdude** [comment]
Just because you can upload your program using a programmer (USBasp in your case) doesn't necessarily means the bootloader is installed.

Try selecting **1 MHz internal** and hit **Burn Bootloader**. the 1 MHz bootloader uses a lower baud rate, and may be more reliable.

**Comment by janioanselmo** [comment]
I just repeat the whole process!
Follow the video as proof! However, still unsuccessful for recording via UART / Serial. I'll check your latest information! Thank you.
[Burn Bootloader - ATmega2561.zip](https://github.com/MCUdude/MegaCore/files/1940243/Burn.Bootloader.-.ATmega2561.zip)

**Comment by janioanselmo** [comment]
Hey @MCUdude, I found out where I went wrong! thanks YOUR HELP! What happens is, I did not put the LED on the 13 pin. And whenever I was recording I expected the "LED flashing". And now that I've installed, I've been waiting for the correct installation time. Well, when I used USBasp, to burn the bootloader, I did not know the correct time. That is, I disconnected as soon as I received "OK" from the bootloader recording, and with the LED, I was able to wait for the correct time! Anyway, THANK YOU for the patience and attention you are a reference for us amateurs! Thank you again!
